### PR TITLE
Add Module Structure

### PR DESCRIPTION
**Pull request by tylerbwong**
Also adds a new `:cli` module.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,65 +1,13 @@
...
+    mavenLocal()
+    maven("https://plugins.gradle.org/m2/")
+    jcenter()
+    maven("https://kotlin.bintray.com/ktor")
+  }
 }
 
 task("buildReact", Exec::class) {
-  workingDir("$workingDir/web")
-  inputs.dir("$workingDir/src")
```
Comment (cdubthecoolcat) [inline build.gradle.kts:51]: This line is needed for the continuous build feature of Gradle.

File: build.gradle.kts [hunk 2]
```diff
@@ -1,65 +1,19 @@
...
-  implementation("org.jetbrains.exposed", "exposed-java-time", exposedVersion)
-  implementation("org.postgresql", "postgresql", postgresqlVersion)
-  testImplementation("io.ktor", "ktor-server-tests", ktorVersion)
 }
 
 task("buildReact", Exec::class) {
-  workingDir("$workingDir/web")
-  inputs.dir("$workingDir/src")
+  workingDir("$rootDir/web")
+  inputs.dir("$rootDir/server/src")
```
Comment (cdubthecoolcat) [inline build.gradle.kts:17]: This should be `inputs.dir("$rootDir/web/src")`

File: server/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  implementation("io.ktor", "ktor-serialization", ktorVersion)
+  implementation("org.jetbrains.exposed", "exposed-core", exposedVersion)
+  implementation("org.jetbrains.exposed", "exposed-dao", exposedVersion)
+  implementation("org.jetbrains.exposed", "exposed-jdbc", exposedVersion)
+  implementation("org.jetbrains.exposed", "exposed-java-time", exposedVersion)
+  implementation("org.postgresql", "postgresql", postgresqlVersion)
+  testImplementation("io.ktor", "ktor-server-tests", ktorVersion)
+}
+
+task("dockerUp", Exec::class) {
```
Comment (cdubthecoolcat) [inline server/build.gradle.kts:42]: This task fails for me.

File: build.gradle.kts (hunk 2 above)
Comment (tylerbwong) [reply build.gradle.kts]: Whoops, nice catch.

File: server/build.gradle.kts (hunk 3 above)
Comment (tylerbwong) [reply server/build.gradle.kts]: Ah, yea this needs to be a root project task. Should be fixed now.

**Pull request merged**